### PR TITLE
docs(website): add git-hook Japanese translation

### PR DESCRIPTION
**Pull request by OldBigBuddha**
## Summary

Translate [git-hooks](https://biomejs.dev/recipes/git-hooks/) into Japanese

related to #1421

## Test Plan

N/A

### CODE REVIEW

**Comment by unvalley** [comment]
Okay, I'll check

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+- コミットをする前にformatのチェックとlintを行う
+
+  ```yaml title="lefthook.yml"
+  pre-commit:
+    commands:
+      check:
+        glob: "*.{js,ts,cjs,mjs,d.cts,d.mts,jsx,tsx,json,jsonc}"
+        run: npx biome check --no-errors-on-unmatched --files-ignore-unknown=true {staged_files}
+  ```
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:28]: I think this code block is broken. The original does too, so we need to fix it.

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+- コミットをする前にformatチェックやlintを行い、[安全な修正](../linter#安全な修正safe-fixes)を行う
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:30]:
```suggestion
- コミット前にformatチェックやlintを行い、[安全な修正](../linter#安全な修正safe-fixes)を行う
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  pre-commit:
+    commands:
+      check:
+        glob: "*.{js,ts,cjs,mjs,d.cts,d.mts,jsx,tsx,json,jsonc}"
+        run: npx biome check --apply --no-errors-on-unmatched --files-ignore-unknown=true {staged_files} && git update-index --again
+  ```
+
+`git update-index --again` は、ステージングされたファイルを再度ステージングします。
+
+- プッシュする前にformatのチェックとlintを行う
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:42]:
```suggestion
- プッシュ前にformatとlintのチェックを行う
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+  pre-push:
+    commands:
+      check:
+        glob: "*.{js,ts,cjs,mjs,d.cts,d.mts,jsx,tsx,json,jsonc}"
+        run: npx biome check --no-errors-on-unmatched --files-ignore-unknown=true {pushed_files}
+  ```
+
+`glob`と`--files-ignore-unknown=true` を併用する必要はありません。
+`--files-ignore-unknown=true` を指定するだけでBiomeが現在サポートしている、もしくは将来サポートするかもしれないファイルのみを対象とすることができます。
+どのファイルをBiomeの対象にするかをもっとコントロールしたい場合は `glob` を使ってください。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:54]:
```suggestion
`--files-ignore-unknown=true` を指定するだけで、Biomeが現在サポートしている（もしくは将来サポートするかもしれない）ファイルのみを対象にできます。
対象ファイルをさらにコントロールしたい場合は、`glob` を使ってください。
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+`--no-errors-on-unmatched`は、*どのファイルも処理されなかった場合*に発生するエラーを抑制します。
+
+設定したら、`lefthook install` を実行してフックのセットアップを完了させましょう。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:58]:
```suggestion
設定後、`lefthook install` を実行してフックのセットアップを完了させましょう。
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+プロジェクトに `package.json` が含まれている場合、
+パッケージのインストール時に `scripts.postinstall` を使って自動的に _Lefthook_ のフックを設定することができます：
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:61]:
```suggestion
プロジェクトに `package.json` が含まれている場合、パッケージのインストール時に `scripts.postinstall` を使って自動的に _Lefthook_ のフックを設定することができます：
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    "postinstall": "lefthook install"
+  }
+}
+```
+
+
+## Husky
+
+[Husky](https://github.com/typicode/husky)は、JavaScriptエコシステムで広く使われているフックマネージャーです。
+Husky はステージングされていない変更を無視したり、ステージングされているファイルののみを対象とすることができません。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:75]:
```suggestion
Husky はステージングされていない変更を無視したり、ステージングされているファイルのみを対象としたりすることができません。
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+そのため、 _lint-staged_ や _git-format-staged_ といった別のツールと組み合わせて使われることが多いです。
+
+プロジェクトに `package.json` が含まれている場合、
+`scripts.postinstall` を使ってパッケージのインストール時に自動的に _husky_ のフックを設定することができます：
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:79]:
```suggestion
プロジェクトに `package.json` が含まれている場合、`scripts.postinstall` を使ってパッケージのインストール時に自動的に _husky_ のフックを設定することができます：
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+```
+
+*処理されなかったファイルがない* 場合を示すエラーを非表示にするため、コマンド実行時に`--no-errors-on-unmatched`を使用することを忘れないでください。
+
+
+### git-format-staged
+
+_lefthook_、_pre-commit_、_lint-staged_ などの他のツールとは対照的に、[git-format-staged](https://github.com/hallettj/git-format-staged) は内部的に `git stash` を使用しません。
+これにより、ステージングされていない変更とステージングされた変更の間で衝突が発生した場合に手動で介入する必要がなくなります。
+詳しくは [_git-format-staged_ と他のツールの比較](https://github.com/hallettj/git-format-staged#comparisons-to-similar-utilities) を参照してください。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:130]:
```suggestion
_lefthook_、_pre-commit_、_lint-staged_ などの他のツールとは対照的に、[git-format-staged](https://github.com/hallettj/git-format-staged) は内部的に `git stash` を使用しません。
これにより、ステージングされていない変更とステージングされた変更の間で衝突が発生した場合に、手動で介入する必要がなくなります。
詳しくは [_git-format-staged_ と他のツールの比較](https://github.com/hallettj/git-format-staged#comparisons-to-similar-utilities) を参照してください。
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+
+
+設定例：
+
+- コミットをする前にformatのチェックとlintを行う
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:135]:
```suggestion
- コミット前にformatとlintのチェックを行う
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+  ```shell title=".husky/pre-commit"
+  git-format-staged --formatter 'biome check --files-ignore-unknown=true --no-errors-on-unmatched \"{}\"' .
+  ```
+
+- コミットをする前にformatチェックやlint、安全な修正を行う
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:140]:
```suggestion
- コミット前にformatやlintのチェック・安全な修正を行う
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+
+  ```shell title=".husky/pre-commit"
+  git-format-staged --formatter 'biome check --apply --files-ignore-unknown=true --no-errors-on-unmatched \"{}\"' .
+  ```
+
+## pre-commit
+
+
+[pre-commit](https://pre-commit.com/)は多言語フックマネージャーです。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:149]:
```suggestion
[pre-commit](https://pre-commit.com/)は、多言語フックマネージャーです。
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+    rev: "0.1.0"  # タグやハッシュ値を指定してください
+    hooks:
+    -   id: biome-check
+        additional_dependencies: ["@biomejs/biome@1.4.1"]
+```
+
+これで`git commit`を実行したときに `biome check --apply` が実行されるようになりました。
+
+[pre-commit](https://pre-commit.com/)は別々にツールをインストールするため、pre-commit はどのツールがインストールされるか知っておく必要があります。
+なので `additional_dependencies` オプションを使ってBiomeのバージョンを指定しなくてはいけません。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:176]:
```suggestion
[pre-commit](https://pre-commit.com/)は別々にツールをインストールするため、pre-commit はどのツールがインストールされるか知っておく必要があります。
そのため、`additional_dependencies` オプションを使ってBiomeのバージョンを指定しなくてはいけません。
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 14]
```diff
@@ -0,0 +1,235 @@
...
+
+Biome がすでに `npm` のパッケージとしてローカルリポジトリにインストールされている場合、
+Biome を更新する際に `package.json` と `.pre-commit-config.yaml` の両方を更新するのは開発者の負担になります。
+提供されているBiomeフックを使う代わりに、独自の[ローカルフック](https://pre-commit.com/index.html#repository-local-hooks)を指定することができます。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:180]:
```suggestion
Biome がすでに `npm` のパッケージとしてローカルリポジトリにインストールされている場合、Biome を更新する際に `package.json` と `.pre-commit-config.yaml` の両方を更新するのは開発者の負担になります。
提供されているBiomeフックを使う代わりに、独自の[ローカルフック](https://pre-commit.com/index.html#repository-local-hooks)を指定できます。
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+      - id: local-biome-check
+        name: biome check
+        entry: npx biome check --apply --files-ignore-unknown=true --no-errors-on-unmatched
+        language: system
+        types: [text]
+        files: "\\.(jsx?|tsx?|c(js|ts)|m(js|ts)|d\\.(ts|cts|mts)|jsonc?)$"
+```
+
+Biome は`--files-ignore-unknown=true`を指定することで未知のファイルを無視するため、
+pre-commit の `files` オプションは必須ではありません。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:197]:
```suggestion
Biome は`--files-ignore-unknown=true`を指定することで未知のファイルを無視するため、pre-commit の `files` オプションは必須ではありません。
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+## Shell script
+
+独自のシェルスクリプトを使うこともできます。
+クロスプラットフォームの互換性に関する問題に遭遇する可能性があることに注意してください。
+前のセクションで紹介したような専用ツールの使用をおすすめします。
+
+
+シェルスクリプトの例：
+
+- コミットをする前にformatのチェックとlintを行う
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:208]:
```suggestion
- コミット前にformatとlintのチェックを行う
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 17]
```diff
@@ -0,0 +1,235 @@
...
+
+  ```shell title=".git/hooks/pre-commit"
+  #!/bin/sh
+  set -eu
+
+  git diff -z --staged --name-only --diff-filter=ACMR | \
+      xargs -0 npx biome check --files-ignore-unknown=true --no-errors-on-unmatched $STAGED_FILES
+  ```
+
+- コミットをする前にformatチェックやlint、安全な修正を行う
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:218]:
```suggestion
- コミット前にformatやlintのチェック、安全な修正を行う
```

File: website/src/content/docs/ja/recipes/git-hooks.mdx [hunk 18]
```diff
@@ -0,0 +1,235 @@
...
+    exit 1;
+  fi
+
+  git diff -z --staged --name-only --diff-filter=ACMR | \
+      xargs -0 npx biome check --files-ignore-unknown=true --no-errors-on-unmatched
+
+  git update-index --again
+  ```
+
+すでにステージングされたファイルに新たな変更がある場合はフックを失敗させていることに注意してください。
```
Comment (unvalley) [inline website/src/content/docs/ja/recipes/git-hooks.mdx:235]:
```suggestion
すでにステージングされたファイルに新たな変更がある場合は、フックを失敗させていることに注意してください。
```

**Review by unvalley — CHANGES_REQUESTED** [review]
Thank you for contribution! I left some minor suggestions.

**Comment by OldBigBuddha** [comment]
@unvalley I appreciate your reviews. I applied all your suggestions!
Please check it again 🙏

**Comment by unvalley** [comment]
@OldBigBuddha Thank you, looks good to me!

**Review by unvalley — APPROVED** [review]

**Pull request merged**